Fails to compile in Xcode 9

With Swift version 3.2 as compiled with Xcode 9 - Beta 1.
Installed as a pod with use_framworks!

```
/Users/geoffmacdonald/imgur-ios/Pods/Eureka/Source/Core/Section.swift:206:1: error: 'Collection' requires the types 'Section.SubSequence' (aka 'Array<BaseRow>') and 'ArraySlice<BaseRow>' be equivalent
extension Section : MutableCollection, BidirectionalCollection {
^
/Users/geoffmacdonald/imgur-ios/Pods/Eureka/Source/Core/Section.swift:206:1: note: requirement specified as 'Self.SubSequence' == 'Self.SubSequence.SubSequence' [with Self = Section]
extension Section : MutableCollection, BidirectionalCollection {
^
/Users/geoffmacdonald/imgur-ios/Pods/Eureka/Source/Core/SelectableSection.swift:65:43: error: associated type 'Self.SubSequence.SubSequence' cannot be equal to both 'Section.SubSequence' (aka 'Array<BaseRow>') and 'ArraySlice<BaseRow>'
extension SelectableSectionType where Self: Section, Self.Iterator == IndexingIterator<Section>, Self.Iterator.Element == BaseRow {
                                          ^
/Users/geoffmacdonald/imgur-ios/Pods/Eureka/Source/Core/Section.swift:206:1: error: 'Collection' requires the types 'Section.SubSequence' (aka 'Array<BaseRow>') and 'ArraySlice<BaseRow>' be equivalent
extension Section : MutableCollection, BidirectionalCollection {
^
/Users/geoffmacdonald/imgur-ios/Pods/Eureka/Source/Core/Section.swift:206:1: note: requirement specified as 'Self.SubSequence' == 'Self.SubSequence.SubSequence' [with Self = Section]
extension Section : MutableCollection, BidirectionalCollection {
```

--- a/Source/Core/Section.swift
+++ b/Source/Core/Section.swift
@@ -232,8 +232,8 @@ extension Section : MutableCollection, BidirectionalCollection {
         }
     }
 
-    public subscript (range: Range<Int>) -> [BaseRow] {
-        get { return kvoWrapper.rows.objects(at: IndexSet(integersIn: range)) as! [BaseRow] }
+    public subscript (range: Range<Int>) -> ArraySlice<BaseRow> {
+        get { return ArraySlice<BaseRow>(kvoWrapper.rows.objects(at: IndexSet(integersIn: range)) as! [BaseRow]) }
         set {
             replaceSubrange(range, with: newValue)
         }

